ID: 19713
Title: Unit test `TestAutoUpdate` failed in 4.0 branch when new collation is enabled.
Description:
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Enable the new collation by changing the code in 

```
func init() {
        newCollationEnabled = 1  // always enable the new collation

	newCollatorMap = make(map[string]Collator)
        ...
```

And then run the test `TestAutoUpdate`:

```
bash$ cd $TIDB_SRC/statistics/handle
bash$ go test -check.f TestAutoUpdate
```

### 2. What did you expect to see? (Required)
The test case ran successfully.

### 3. What did you see instead (Required)
```
FAIL: update_test.go:371: testStatsSuite.TestAutoUpdate

update_test.go:464:
    c.Assert(hg.NDV, Equals, int64(3))
... obtained int64 = 1
... expected int64 = 3
```

### 4. What is your TiDB version? (Required)
The commit hash is `3b2c7e763a3a918c5247fb2ac7cf551cb0460679` in `release-4.0`.

This issue is not reproduced in `master` branch.